fetch_index.py

```diff
@@ -42,20 +66,78 @@ def __init__(self, data, query):
         self.sharepoint_host = data.get("sharepoint.host_url")
         self.objects = data.get("objects")
         self.site_collections = data.get("sharepoint.site_collections")
-        self.query = query
-        self.start_time = data.get("start_time")
-        self.end_time = data.get("end_time")
+        self.enable_permission = data.get("enable_document_permission")
+        self.start_time = start_time
+        self.end_time = end_time
         self.checkpoint = Checkpoint(logger, data)
         self.sharepoint_client = SharePoint(logger)
         self.permissions = Permissions(logger, self.sharepoint_client)
         self.ws_client = WorkplaceSearch(self.ws_host, http_auth=self.ws_token)
-
-    def index_sites(self, collection, index):
+        self.mapping_sheet_path = data.get("sharepoint_workplace_user_mapping")
+
+    def index_document(self, document, success_message, failure_message, param_name):
+        """ This method indexes the documents to the workplace.
+            :param document: document to be indexed
+            :param success_message: success message
+            :param failure_message: failure message while indexing the document
+            :param param_name: parameter name whether it is SITES, LISTS OR ITEMS
+        """
+        try:
+            if document:
+                total_documents_indexed = 0
+                document_list = [document[i * DOCUMENT_SIZE:(i + 1) * DOCUMENT_SIZE] for i in range((len(document) + DOCUMENT_SIZE - 1) // DOCUMENT_SIZE)]
```

Comment:
This line would partition the large list of documents in chunks of 100 so that we can do batch indexing(with 100 documents/req) in the workplace. The variable `i` is defined in the same line @ `for i in range((len(document) + DOCUMENT_SIZE - 1)) ...`